pkg/controller/controller.go

```diff
@@ -312,6 +313,7 @@ func (c *Controller) Reconcile(req reconcile.Request) (reconcile.Result, error)
 		c.logger.Errorf(ctx, err, "failed to reconcile")
 		return reconcile.Result{}, nil
 	}
+	reportLastReconciled(obj)
```

Comment:
Operators would report `last_reconciled` metrics only if they finished reconciliation successfully. That's why I don't like to use `defer` here.